.github/workflows/toolchain-version-check.yaml

```diff
@@ -0,0 +1,63 @@
+---
+name: Toolchain Version Check
+"on":
+  schedule:
+    - cron: "0 0 * * *" # build nightly!
+
+jobs:
+  check-toolchain-version:
+    name: Nightly
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4.1.1
+
+      - name: Install Ruby toolchain
+        uses: ruby/setup-ruby@8575951200e472d5f2d95c625da0c7bec8217c42 # v1.161.0
+        with:
+          ruby-version: ".ruby-version"
+          bundler-cache: true
+
+      - name: Read existing rustc version
+        id: rust_version
+        run: |
+          bundle exec rake toolchain:version
+          echo "version=$(bundle exec rake toolchain:version)" >> "$GITHUB_OUTPUT"
+
+      - name: Check the latest rustc version
+        id: latest_rust_release
+        uses: octokit/request-action@v2.1.9
```

Comment:
let's use a SHA to pin the version. We only use tag versions for GitHub actions in the @actions and @artichoke GitHub organizations.

Reply:
I was curious why some where SHA's. I assume to fix any risk of supply chain attacks and floating tag issues. I'm curious, does it work with dependabot?

Reply:
yes it does. Dependabot will also update the comment which indicates the tag.